### DISCUSSION TITLE
ARM64 support?

### DISCUSSION THREAD

**Discussion by liufsd** [abbeycode/UnrarKit#96] [Q&A]
Showing All Messages
The linked and embedded framework 'UnrarKit.framework' is missing one or more architectures required by this target: arm64.

**Comment by abbeycode** [comment]
How are you installing UnrarKit? Are you using Carthage or CocoaPods, or downloading the framework from GitHub?

**Reply by liufsd** [reply]
install by Carthage.
build success, but archive fail.
archive required arm64!

**Reply by abbeycode** [reply]
What version of Carthage are you using? I have automated builds that are supposed to be testing Carthage installation, and they're succeeding. I'm trying to rule out causes.

**Reply by liufsd** [reply]
Carthage build success and run it as work，but fail when i archive project…

**Comment by liufsd** [comment]
Mac os app.

Archive fail:

![image](https://user-images.githubusercontent.com/1910405/107482443-a918dc80-6bba-11eb-940d-7b711b4111e6.png)

**Reply by abbeycode** [reply]
That message is coming from Xcode when you do a Build and Archive operation?

**Reply by liufsd** [reply]
yes

**Reply by liufsd** [reply]
build ok ，but archive fail by this message， maybe archive must be support M1 ，but build just check Intel CPU？

**Comment by abbeycode** [comment]
So, here's what I've found:

1. Xcode 12 requires .xcframework bundles for multi-architecture Mac apps
2. I upload binaries for each release of UnrarKit, and (according to Carthage/Carthage#3130) binary downloads don't yet support xcframeworks
3. Due to a bug/quirk of Carthage (which I reported in Carthage/Carthage#3134), even if you pass the `--use-xcframeworks` flag like Carthage is recommending for Xcode 12, it will download the .framework instead and cause the issue you're seeing

As a workaround, you can update or build in Carthage with these two flags, and you should be good:

    carthage update --use-xcframeworks --no-use-binaries

At some point once xcframework downloading/archiving is supported, I'll update to post them instead of, or in addition to, the .framework bundles (I'll post both if it's supported).

**Reply by liufsd** [reply]
sorry  , still not working.

➜   carthage update --use-xcframeworks --no-use-binaries --platform macOS
*** Fetching UnrarKit
*** Checking out UnrarKit at "2.9"
*** xcodebuild output can be found in /var/folders/x2/2k42tnbx2pn4tmr2thq0ybch0000gn/T/carthage-xcodebuild.0rAt6t.log
*** Building scheme "UnrarKit" in UnrarKit.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace ../Carthage/Checkouts/UnrarKit/UnrarKit.xcworkspace -scheme UnrarKit -configuration Release -derivedDataPath .../Library/Caches/org.carthage.CarthageKit/DerivedData/12.4_12D4e/UnrarKit/2.9 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive VALIDATE_WORKSPACE=NO -archivePath /var/folders/x2/2k42tnbx2pn4tmr2thq0ybch0000gn/T/UnrarKit SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in ../Carthage/Checkouts/UnrarKit)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/x2/2k42tnbx2pn4tmr2thq0ybch0000gn/T/carthage-xcodebuild.0rAt6t.log

**Reply by abbeycode** [reply]
You need to use the v2.10 branch until it's merged to `master`

**Reply by liufsd** [reply]
work ! 
~ thanks.

**Comment by lcandy2** [comment]
Same situation, and trying to fix it but failed.

Cartfile
`github "abbeycode/UnrarKit"`

Terminal
`> carthage update --use-xcframeworks --no-use-binaries
*** Fetching UnrarKit
*** Checking out UnrarKit at "2.10"
*** xcodebuild output can be found in /var/folders/31/6xkyyrvj3fd1dhjkxdqn192m0000gn/T/carthage-xcodebuild.DU43CZ.log
*** Building scheme "UnrarKit" in UnrarKit.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace ../Carthage/Checkouts/UnrarKit/UnrarKit.xcworkspace -scheme UnrarKit -configuration Release -derivedDataPath /~/Library/Caches/org.carthage.CarthageKit/DerivedData/14.3.1_14E300c/UnrarKit/2.10 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive VALIDATE_WORKSPACE=NO -archivePath /var/folders/31/6xkyyrvj3fd1dhjkxdqn192m0000gn/T/UnrarKit SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in ../Carthage/Checkouts/UnrarKit)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/31/6xkyyrvj3fd1dhjkxdqn192m0000gn/T/carthage-xcodebuild.DU43CZ.log`

**Reply by lcandy2** [reply]
everything solved using pr98 branch

Cartfile
`github "abbeycode/UnrarKit" "pr98"`
